src/services/codefixes/removeUnnecessaryAwait.ts

```diff
@@ -27,12 +27,34 @@ namespace ts.codefix {
         }
 
         const parenthesizedExpression = tryCast(awaitExpression.parent, isParenthesizedExpression);
-        // (await 0).toFixed() should keep its parens (or add an extra dot for 0..toFixed())
-        if (parenthesizedExpression && isPropertyAccessExpression(parenthesizedExpression.parent) && isDecimalIntegerLiteral(awaitExpression.expression)) {
+        const preserveParens = parenthesizedExpression && (

```

Comment:
I still think this is the right thing to do, but now that we have 3 exceptions, the queasy feeling in my stomach says that there more out there. I *almost* would like to always preserve parentheses, except that the result is so ugly in the common case.

What if we flipped it to only skip parens for identifiers and callexpressions?

Reply:
Yeah, I had the same feeling. I think this is probably a good plan.